documentation/collectionruleexamples.md

```diff
@@ -0,0 +1,223 @@
+# Collection Rule Examples
+
+The following examples provide sample scenarios for using a collection rule. These templates can be copied directly into your configuration file with minimal adjustments to work with your application, or they can be adjusted for your specific use-case. [Learn more about configuring collection rules](collectionrules.md).
+
+## Collect GCDump - Startup (`Startup` Trigger)
+
+### JSON
+
+```json
+{
+  "GCDumpOnStartup": {
+    "Trigger": {
+      "Type": "Startup"
+    },
+    "Actions": [
+      {
+        "Type": "CollectGCDump",
+        "Settings": {
+          "Egress": "artifacts"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Explanation
+
+This rule, named "GCDumpOnStartup", will trigger on a process's startup. When the rule is triggered, a GCDump will be collected and egressed to the specified `Egress` provider (in this case, `artifacts` has been configured to save the GCDump to the local filesystem).
+
+## Collect Trace - High CPU Usage (`EventCounter` Trigger)
+
+### JSON
+
+```json
+{
+  "HighCpuUsage": {
+    "Trigger": {
+      "Type": "EventCounter",
+      "Settings": {
+        "ProviderName": "System.Runtime",
+        "CounterName": "cpu-usage",
+        "GreaterThan": 60,
+        "SlidingWindowDuration": "00:00:10"
+      }
+    },
+    "Actions": [
+      {
+        "Type": "CollectTrace",
+        "Settings": {
+          "Profile": "Cpu",
+          "Egress": "artifacts"
+        }
+      }
+    ],
+    "Filters": [
+      {
+        "Key": "ProcessName",
+        "Value": "MyProcessName"
+      }
+    ]
+  }
+}
+```
+
+### Explanation
+
+This rule, named "HighCpuUsage", will trigger when a process named "MyProcessName" causes CPU usage to exceed 60% for greater than 10 seconds. If the rule is triggered, a Cpu trace will be collected for the default duration (30 seconds), and egressed to the specified `Egress` provider (in this case, `artifacts` has been configured to save the trace to the local filesystem). There is a default `ActionCount` limit stating that this rule may only be triggered 5 times.
+
+## Collect Dump - 4xx Response Status (`AspNetResponseStatus` Trigger)
+
+### JSON
+
+```json
+{
+  "BadResponseStatus": {
+    "Trigger": {
+      "Type": "AspNetResponseStatus",
+      "Settings": {
+        "ResponseCount": 5,
+        "StatusCodes": [
+          "400-499"
```

Comment:
Yep, they do, but when I tested this it actually required 7 404s to trigger it (which is in-line with what I found in the bug, since it took 3 404s to trigger when the `ResponseCount` was supposed to be 1)